### PR TITLE
Remove resource injection on the node for container task

### PR DESCRIPTION
**Pull request by ByronHsu**
# TL;DR

Remove resource injection on the node for container task

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

In [BuildRawContainer](https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/flytek8s/container_helper.go#L213), we do not inject the container resource from the  task template. Instead, we inject the resource from [parameters.TaskExecMetadata.GetOverrides().GetResources()](https://github.com/flyteorg/flytepropeller/blob/master/pkg/compiler/transformers/k8s/node.go#L49).

This is because for both `_with_override(resource=...` and `@task(requests=...)`, they are both inject as the node resource override in [buildNodeSpec](https://github.com/flyteorg/flytepropeller/blob/master/pkg/compiler/transformers/k8s/node.go#L49)

### Expected behavior

For the case of `@task(requests=...)`, the resource override on the node should be empty, and the resource is injected from tasktemplate at [BuildRawContainer](https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/flytek8s/container_helper.go#L213)

## Tracking Issue

https://github.com/flyteorg/flyte/issues/3540

### CODE REVIEW

**Comment by EngHabu** [comment]
I'm probably missing something but should we be trying to do that here? https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/flytek8s/container_helper.go#L305

What issue are you seeing, @ByronHsu ?

**Comment by ByronHsu** [comment]
@EngHabu Thanks for the quick reply. Yes, I have another PR injecting the resource there. See the attachment in the PR https://github.com/flyteorg/flyte/issues/3540.
No issue for the OSS version, but we are hacking something internally, and this logic flaw is blocking us.

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/benlaurie/objecthash v0.0.0-20180202135721-d1e3d6079fc1
 	github.com/fatih/color v1.13.0
 	github.com/flyteorg/flyteidl v1.3.14
-	github.com/flyteorg/flyteplugins v1.0.45
+	github.com/flyteorg/flyteplugins v1.0.48
```
Comment (hamersaw) [inline go.mod:10]: this should be 1.0.47 - https://github.com/flyteorg/flyteplugins/releases. 1.0.48 will pull in code deprecating the barrier transitions, which is not meant to be a part of this merge.

Comment (ByronHsu) [reply go.mod]: done. thanks for the catch

**Review by hamersaw — APPROVED** [review]
